imap/http_caldav.c

```diff
@@ -8220,10 +8220,10 @@ static int meth_options_cal(struct transaction_t *txn, void *params)
                    txn->req_tgt.path);
         strarray_appendm(&txn->resp_body.links, buf_release(&link));
     }
-    if (txn->req_tgt.allow & ALLOW_CAL_SCHED) {
+    if ((txn->req_tgt.allow & ALLOW_CAL_SCHED) && txn->req_tgt.mbentry) {
         struct buf temp = BUF_INITIALIZER;
         if (!annotatemore_lookup_mbe(txn->req_tgt.mbentry, DAV_ANNOT_NS "<" XML_NS_CYRUS ">scheduling-enabled", "", &temp)
-            && (!strcasecmp(buf_cstring(&temp), "F") || !strcasecmp(temp.s, "no")))
+            && (!strcasecmp(buf_cstring(&temp), "F") || !strcasecmp(buf_cstring(&temp), "no")))
```

Comment:
This is not needed.  The first call of buf_cstring(&temp) has to ensure that temp.s is `\0`-terminated.  The second call, added here, will allocate one more byte, if not yet allocated, for a `\0` termination, irrespective on whether the temp.s already ends with `\0`.